File: app/src/main/java/com/juniperphoton/myerlist/presenter/MainContract.java
package com.juniperphoton.myerlist.presenter;

import com.juniperphoton.myerlist.model.ToDo;

public class MainContract {
    public interface View {
        void displayCategories();

        void displayToDos();

        void uploadOrders();

        void notifyToDoDeleted(int pos);
    }

    public interface Presenter extends com.juniperphoton.myerlist.presenter.Presenter {
        void getCates();

        void getToDos();

        void updateOrders(String order);

        void updateIsDone(ToDo toDo);

        void modifyToDo(final String cate, final String content, final String id);

        void addToDo(String cate, String content);

        void deleteToDo(ToDo toDo);

        void clearDeletedList();

        void recoverToDo(ToDo toDo);
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/presenter/MainContract.kt
package com.juniperphoton.myerlist.presenter

import com.juniperphoton.myerlist.model.ToDo

class MainContract {
    interface View {
        fun refreshCategoryList()

        fun refreshToDoList(filter: Int)

        fun notifyDataSetChanged()

        fun uploadOrders()

        fun toggleRefreshing(show: Boolean)

        fun reCreateView()

        fun updateFilterIcon(filterOption: Int)
    }

    interface Presenter : com.juniperphoton.myerlist.presenter.Presenter {
        var filterOption: Int

        fun getCategories()

        fun getToDos()

        fun updateOrders(order: String)

        fun updateIsDone(id: String, oldValue: String)

        fun modifyToDo(id: String, cate: String, content: String)

        fun addToDo(cate: String, content: String)

        fun deleteToDo(item: ToDo)

        fun clearDeletedList()

        fun recoverToDo(toDo: ToDo)

        fun onCategorySelected(item: Int)
    }
}
